File: leetcode_314.cpp
class Solution {
public:
  string simplifyPath(string const& path) {
    vector<string> dirs;

    for (auto i = path.begin(); i != path.end(); ) {
      ++i;

      auto j = find(i, path.end, '/');
      auto dir = string(i, j);

      if (!dir.empty() && dir != ".") {
        if (dir == "..") {
          if (!dirs.empty()) {
            dirs.pop_back();
          }
        } else {
          dirs.push_back(dir);
        }
      }

      i = j;
    }

    stringstream out;
    if (dirs.empty()) {
      out << "/";
    } else {
      for (auto dir : dirs) {
        out << '/' << dir;
      }
    }

    return out.str();
  }
}

File: leetcode_314.py
class Solution:
    # @param path, a string
    # @return a string
    def simplifyPath(self, path):
        splits = path.split('/')
	result = []
	for s in splits:
	  if s == '' or s == '.':
	    pass
	  elif s == '..' and len(result) > 0:
	    result.pop()
	  elif s != '..':
	    result.append(s)
	return '/' + '/'.join(result)


s = Solution()
print s.simplifyPath('/a/./b/../../c/')
print s.simplifyPath('/..')
